src/android/ContactManager.java

```diff
@@ -48,13 +54,66 @@ Licensed to the Apache Software Foundation (ASF) under one
     public static final int NOT_SUPPORTED_ERROR = 5;
     public static final int PERMISSION_DENIED_ERROR = 20;
     private static final int CONTACT_PICKER_RESULT = 1000;
+    public static String [] permissions;
+
+
+    //Request code for the permissions picker (Pick is async and uses intents)
+    public static final int SEARCH_REQ_CODE = 0;
+    public static final int SAVE_REQ_CODE = 1;
+    public static final int REMOVE_REQ_CODE = 2;
+
 
     /**
      * Constructor.
      */
     public ContactManager() {
+        permissions = new String[2];
+        permissions[0] = Manifest.permission.READ_CONTACTS;
+        permissions[1] = Manifest.permission.WRITE_CONTACTS;
+    }
+
+    public String [] requestPermissions()
+    {
+        return permissions;
```

Comment:
Is every pluign that requires permissions expected to implement this? I understand rest of the code here - but not sure when this will be invoked and used.